core/shared/src/test/scala/fs2/concurrent/TopicSuite.scala

```diff
@@ -173,4 +173,14 @@ class TopicSuite extends Fs2Suite {
       .map(it => assert(it.size <= 11))
     // if the stream won't be discrete we will get much more size notifications
   }
+
+  test("return empty stream if closed") {
+    val program = for {
+      topic <- Topic[IO, Int]
+      _ <- Stream(1, 2, 3).through(topic.publish).compile.drain
+      _ <- topic.subscribeUnbounded.compile.drain
+    } yield ()
+    
+    program.timeout(5.seconds) // will timeout if program is deadlocked
```

Comment:
Ok thanks. `executeEmbed` does work. I tried `execute` with `tickAll` and `isDeadlocked` but that didn't work for me so I assumed `executeEmbed` would have the same problem, but apparently not... I'm probably misinterpreting what `tickAll` does then.

Reply:
`isDeadlocked` returns `true` if there is no more work scheduled. 
https://github.com/typelevel/cats-effect/blob/8ba8916407eb8f35fbffae76f72dac48d2dc7357/testkit/shared/src/main/scala/cats/effect/testkit/TestControl.scala#L270-L276

However, this might be because the `results` are already available, which I believe is the case here.
https://github.com/typelevel/cats-effect/blob/8ba8916407eb8f35fbffae76f72dac48d2dc7357/testkit/shared/src/main/scala/cats/effect/testkit/TestControl.scala#L127

I wonder if `isDeadlocked` should check that as well.

Reply:
lol I think I get it. `isDeadlocked` probably also returns true if the program if finished because then there are also no tasks left to execute.